src/Pos/Explorer/Web/Server.hs

```diff
@@ -407,29 +435,32 @@ topsortTxsOrFail f =
     maybeThrow (Internal "Dependency loop in txs set") .
     topsortTxs f
 
+-- | Convert server address to client address,
+-- with the possible exception (effect).
 cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr =
-    fromCAddress cAddr &
-    either (const $ throwM $ Internal "Invalid address!") pure
+cAddrToAddr cAddr = either exception pure (fromCAddress cAddr)
```

Comment:
I dislike point-free. What I wrote is explicit, easier to read and understand. I don't have to look at the types to know what I'm doing.